webchat/home/homeController.js

```diff
@@ -3,11 +3,74 @@
 
     const webchat = angular.module('webchat');
 
-    webchat.controller('HomeController', ['WebchatService', function HomeController (WebchatService) {
+    webchat.controller('HomeController', ['UserService', 'AuthService', 'WebchatService', 'MessageService', '$scope', function HomeController (UserService, AuthService, WebchatService, MessageService, $scope) {
         const homeCtrl = this;
 
-        homeCtrl.contacts = WebchatService.getContacts();
+        homeCtrl.$onInit = () => {
+            homeCtrl.client = AuthService.chatClient;
+            homeCtrl.cachedUsers = {};
 
-    }]);
+            homeCtrl.getUserList(homeCtrl.client.users);
+            homeCtrl.client.on('user-list-updated', homeCtrl.getUserList);
+            homeCtrl.client.on('call-requested', homeCtrl.promptCall);
+            homeCtrl.client.on('chat-created', homeCtrl.chatCreated);
+        };
+
+        homeCtrl.getUserList = (users) => {
+            const parsedUsers = [];
+
+            if (users.forEach) {
```

Comment:
Why?

Reply:
It is used to check if the object has a forEach, because when it is empty, it doesn't have. To avoid this lodash forEach could be used.